keps/sig-cloud-provider/20190704-instance-host-labelling.md

```diff
@@ -0,0 +1,174 @@
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+Many kubernetes clusters are deployed on virtual machines (also referred to as "instances"),
+and information about the physical host for those instances (also referred to as "instance host")
+is not consistently available.
+Instances which share a host can fail simultaneously if there is a problem with the host.
+
+We propose to add a well-known label for nodes to be filled with the instance host ID.
+This label can then be used by controllers (namely, the scheduler) to implement features such as inter-pod affinity and anti-affinity and even pod spreading.
+
+## Motivation
+
+Today, well-known labels for zone and region can be used as topology keys for inter-pod affinity and anti-affinity.
+There is also ongoing work to implement even pod spreading, for which these keys can also be used.
+
+We want to provide more built-in options to improve reliability of workloads running in a kubernetes cluster.
+An instance host can fail and thus disrupting all of its pods, so it should be identified as a topology or failure domain as well.
+
+In addition to reliability, having a well-defined label for the instance host increases the portability of pod specs for customers that have on-prem and cloud deployments.
+
+### Goals
+
+- Introduce a well known label to identify the instance host of a node.
+- Define a default value for the label so that components maintain their current behavior if they use the label.
+
+### Non-Goals
+
+- Add a cloud provider API to get instance host ID.
+- Populate the instance host label with the instance host ID.
+- Controller behavior (e.g. deployment, statefulset) that considers instance host label.
+- Any decision-making that is based on the physical host label.
+
+## Proposal
+
+Add a well-known label for instance host: `topology.kubernetes.io/instance-host` and make its default value empty.
```

Comment:
Don't love the label name `topology.kubernetes.io/instance-host` TBH, I feel it would get confused with just "instance". The alternative is "topology.kubernetes.io/hypervisor` but that is too specific maybe?

Reply:
Some others I've thought of:
* `topology.kubernetes.io/vmm` (virtual machine monitor)
* `topology.kubernetes.io/hardware` 

None of them stick out to me though

Reply:
What about `topology.kubernetes.io/physical-host` (mentioned on #997)? it avoids the confusion and is quite descriptive as well.

Reply:
`topology.kubernetes.io/physical-host` could be a better name IMO.

Reply:
physical-host is not egregious.  The default should not be empty, but the whole label should be undefined.